src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -673,82 +673,56 @@ abstract class CreateOp extends OpCode(0xf0, 3, 1, _.G_create) {
   protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
     val (Seq(endowment, inOffset, inSize), stack1) = state.stack.pop(3)
 
-    val validCall = state.env.callDepth < EvmConfig.MaxCallDepth && endowment <= state.ownBalance
-
-    if (!validCall) {
-      val stack2 = stack1.push(UInt256.Zero)
-      state.withStack(stack2).step()
-    } else {
-
-      val (initCode, memory1) = state.memory.load(inOffset, inSize)
-      val (newAddress, world1) = state.world.createAddressWithOpCode(state.env.ownerAddr)
-      val world2 = world1.initialiseAccount(newAddress).transfer(state.env.ownerAddr, newAddress, endowment)
-
-      // EIP-684
-      val conflict = state.world.nonEmptyCodeOrNonceAccount(newAddress)
-      val code = if (conflict) ByteString(INVALID.code) else initCode
-
-      val newEnv = state.env.copy(
-        callerAddr = state.env.ownerAddr,
-        ownerAddr = newAddress,
-        value = endowment,
-        program = Program(code),
-        inputData = ByteString.empty,
-        callDepth = state.env.callDepth + 1
-      )
-
-      //FIXME: to avoid calculating this twice, we could adjust state.gas prior to execution in OpCode#execute
-      //not sure how this would affect other opcodes [EC-243]
-      val availableGas = state.gas - (constGasFn(state.config.feeSchedule) + varGas(state))
-      val startGas = state.config.gasCap(availableGas)
-
-      val context: ProgramContext[W, S] = state.context.copy(
-        env = newEnv,
-        receivingAddr = newAddress,
-        startGas = startGas,
-        world = world2,
-        initialAddressesToDelete = state.addressesToDelete)
-
-      val result = VM.run(context)
-
-      val contractCode = result.returnData
-      val codeDepositGas = state.config.calcCodeDepositCost(contractCode)
-      val gasUsedInVm = startGas - result.gasRemaining
-      val totalGasRequired = gasUsedInVm + codeDepositGas
-
-      val maxCodeSizeExceeded = state.config.maxCodeSize.exists(codeSizeLimit => contractCode.size > codeSizeLimit)
-      val enoughGasForDeposit = totalGasRequired <= startGas
-
-      val creationFailed = maxCodeSizeExceeded || result.error.isDefined ||
-        !enoughGasForDeposit && state.config.exceptionalFailedCodeDeposit
-
-      if (creationFailed) {
+    //FIXME: to avoid calculating this twice, we could adjust state.gas prior to execution in OpCode#execute
+    //not sure how this would affect other opcodes [EC-243]
```

Comment:
do we still need this comment?

Reply:
Well, it still applies, but I doubt we'll fix it anytime soon